sktime/clustering/spatio_temporal/_st_dbscan.py

```diff
@@ -0,0 +1,296 @@
+"""
+ST-DBSCAN - fast scalable implementation of ST DBSCAN
+            scales also to memory by splitting into frames
+            and merging the clusters together
+"""
+
+__author__ = ["VectorNd"]
+
+from sktime.clustering import BaseClusterer
+
+import numpy as np
+
+from scipy.spatial.distance import pdist, squareform
+from scipy.sparse import coo_matrix
+
+from sklearn.cluster import DBSCAN
+from sklearn.utils import check_array
+from sklearn.neighbors import NearestNeighbors
+
+import warnings
+
+class ST_DBSCAN(BaseClusterer):
+    """
+    A class to perform the ST_DBSCAN clustering
+    Parameters
+    ----------
+    eps1 : float, default=0.5
+        The spatial density threshold (maximum spatial distance) between 
+        two points to be considered related.
+    eps2 : float, default=10
+        The temporal threshold (maximum temporal distance) between two 
+        points to be considered related.
+    min_samples : int, default=5
+        The number of samples required for a core point.
+    metric : string default='euclidean'
+        The used distance metric - more options are
+        'braycurtis', 'canberra', 'chebyshev', 'cityblock', 'correlation',
+        'cosine', 'dice', 'euclidean', 'hamming', 'jaccard', 'jensenshannon',
+        'kulsinski','manhattan'.
+    n_jobs : int or None, default=-1
+        The number of processes to start -1 means use all processors 
+    Attributes
+    ----------
+    labels : array, shape = [n_samples]
+        Cluster labels for the data - noise is defined as -1
```

Comment:
```suggestion
        Cluster labels for the data - noise is defined as -1
        
    References
    ----------
    .. [1] D. Birant and A. Kut, “ST-DBSCAN: An algorithm for clustering spatial-temporal data,” Data Knowl. Eng., vol. 60, no. 1, pp. 208–221, Jan. 2007, doi: [10.1016/j.datak.2006.01.013](https://doi.org/10.1016/j.datak.2006.01.013).
    .. [2] I. Peca, G. Fuchs, K. Vrotsou, N. Andrienko, and G. Andrienko, “Scalable Cluster Analysis of Spatial Events,” 2012, The Eurographics Association. doi: [10.2312/PE/EUROVAST/EUROVA12/019-023](https://doi.org/10.2312/PE/EUROVAST/EUROVA12/019-023).
    ...[3] E. Cakmak, M. Plank, D. S. Calovi, A. Jordan, and D. Keim, “Spatio-temporal clustering benchmark for collective animal behavior,” in Proceedings of the 1st ACM SIGSPATIAL International Workshop on Animal Movement Ecology and Human Mobility, Beijing China: ACM, Nov. 2021, pp. 5–8. doi: [10.1145/3486637.3489487](https://doi.org/10.1145/3486637.3489487).
```

Reply:
@vagechirkov, is this all the same algorithm? Maybe a more specific pointer (section etc) would be appropriate, if some of these are review papers. Also, there seem to be an extra dot in front of the 3.

Reply:
The first paper presents the algorithm, the second paper is the extension that describes the splitting of the data into frames (it is more relevant for the `_fit_frame_split` method). The third paper presents the benchmark and the code, which is almost identical to the current implementation. I guess I can point to the methods and results section in the third paper, but the paper is 4 pages long 🤷‍♂️